extensions/attributegen/plugin_test.cc

```diff
@@ -0,0 +1,423 @@
+}
+
+TEST_P(AttributeGenFilterTest, ExprEvalError) {
+  const char* plugin_config = R"EOF(
+                    {"attributes": [{"output_attribute": "istio.operationId",
+                    "match": [{"value":
+                            "GetStatus", "condition": "request.url_path"}]}]}
+  )EOF";
+  setupConfig({.plugin_config = plugin_config});
+
+  Http::TestRequestHeaderMapImpl request_headers{{":path", "/status/207"}};
+  Http::TestResponseHeaderMapImpl response_headers{{":status", "404"}};
+
+  makeTestRequest(request_headers, response_headers);
+
+  auto fs = request_stream_info_.filterState();
+  ASSERT_EQ(fs->hasData<WasmState>("istio.operationId"), false);
+  ASSERT_EQ(fs->hasData<WasmState>("istio.operationId_error"), true);
+}
+
+TEST_P(AttributeGenFilterTest, UnparseableConfig) {
+  const char* plugin_config = R"EOF(
+                    attributes = [ output_attribute ]; 
+  )EOF";
+  EXPECT_THROW_WITH_MESSAGE(setupConfig({.plugin_config = plugin_config}),
+                            EnvoyException, "Configuration failed");
+}
+
+TEST_P(AttributeGenFilterTest, BadExpr) {
+  const char* plugin_config = R"EOF(
+                    {"attributes": [{"output_attribute": "istio.operationId",
+                    "match": [{"value":
+                            "GetStatus", "condition": "if a = b then return 5"}]}]}
+  )EOF";
+  EXPECT_THROW_WITH_MESSAGE(setupConfig({.plugin_config = plugin_config}),
+                            EnvoyException, "Configuration failed");
+}
+
+TEST_P(AttributeGenFilterTest, NoMatch) {
+  const char* plugin_config = R"EOF(
+                    {"attributes": [{"output_attribute": "istio.operationId",
+                    "match": [{"value":
+                            "GetStatus", "condition": "request.url_path.startsWith('/status') && request.method == 'POST'"}]}]}
+  )EOF";
+  setupConfig({.plugin_config = plugin_config});
+
+  Http::TestRequestHeaderMapImpl request_headers{{":path", "/status/207"},
+                                                 {":method", "GET"}};
+  Http::TestResponseHeaderMapImpl response_headers{{":status", "404"}};
+
+  makeTestRequest(request_headers, response_headers);
+
+  auto fs = request_stream_info_.filterState();
+  ASSERT_EQ(fs->hasData<WasmState>("istio.operationId"), false);
+  ASSERT_EQ(fs->hasData<WasmState>("istio.operationId_error"), false);
+}
+
+TEST_P(AttributeGenFilterTest, OperationFile) {
+  const std::string attribute = "istio.operationId";
+
+  setupConfig(
+      {.plugin_config_file = "operation.json"});  // testdata/operation.json
+
+  Http::TestRequestHeaderMapImpl request_headers{{":path", "/books"},
+                                                 {":method", "GET"}};
+  Http::TestResponseHeaderMapImpl response_headers{{":status", "404"}};
+
+  verify_request(request_headers, response_headers, attribute, true, false,
+                 "ListBooks");
+}
+
+// CEL is unable to get access to response status code.
```

Comment:
@kyessenov @jplevyak CEL is unable to get access to `response.code` .
This works fine in an actual full manual verification, but does not work in this test.
Setting response header `{":status", "404"}` is not populating response.code internally.